cmd/docgen/completion/completion.go

```diff
@@ -26,7 +27,7 @@ func NewCompletion(types []Type, root []*Property) *Completion {
 	// extract types which are available in a context without a session
 	rootNoSession := make([]*Property, 0, len(rootNoSessionTypes))
 	for _, p := range root {
-		if rootNoSessionTypes[p.Type] {
+		if rootNoSessionTypes.Contains(p.Type) {
```

Comment:
Feel a little like i'm cheating in go here but I find the "set of strings" such a common pattern.. and this just a little bit more readable

Reply:
I don't care that much but I actually think there's far more value to sticking to the idiomatic way of doing it instead of creating a cutesy wrapper. Because now I have to wonder whether that wrapper does anything else different. Nobody who uses go is going to have any problem understanding the map version immediately.

Reply:
I bit the bullet here because sometimes you also need to end up with the strings as a slice and the regular go way doesn't do that.. getting keys from maps is one of my major gripes with go

Reply:
Wouldn't a single function satisfy that if you are doing it often? I don't know, just feels like trying to make Go into Java. :)

Reply:
Maybe I dunno.. ok I'll ditch this.. and add `StringMapKeys` function